### ISSUE TITLE
dnf5daemon search (org.rpm.dnf.v0.rpm.Rpm list) pattern into summary and description

### ISSUE THREAD

**Issue by anaselli** [rpm-software-management/dnf5#1142]
The dnfdaemon based on dnf4 allows to search packages by looking for strings contained into summary or description attributes. 
It seems that is not possible any more.

**Comment by ppisar** [comment]
I confirm this deficiency on dnf5daemon-server-5.1.10-20240108005144.10.g4c473346.fc40.x86_64. Now org.rpm.dnf.v0.rpm.Rpm interface only supports:

~~~~
            - with_nevra: bool (default true)
                match patterns against available packages NEVRAs
            - with_provides: bool (default true)
                match patterns against available packages provides
            - with_filenames: bool (default true)
                match patterns against names of the files in available packages
            - with_binaries: bool (default true)
                match patterns against names of the binaries in `/usr/(s)bin` in available packages
            - with_src: bool (default true)
                include source rpms into the results
~~~~

There is missing a boolean to search in summaries or descriptions.

**Comment by m-blaha** [comment]
Right. I'm only not sure whether this "search" capability should be part of the `list()` call or whether we should rather create a new `search()` method.

**Comment by ppisar** [comment]
Do you mean to distinct between RPM dependencies and other RPM headers? Summary and Description are a free text which is not used for resolving dependencies. I can imagine one could also search in License. On the other hand, a hypothetical search in Conflicts could fall into the current `list()` method. I think this classification could become blurry. And in my opinion it is not substantial.

I wouldn't create a new function if not necessary.

One reason for a separate function could be a need for a completely different implementation: e.g. if `list()` were implemented with libsolv and thus subjected to libsolv constrains and semantics, while `search()` would be implemented on libdnf5 level.

Other reason for a separate function could a complex query: I want packages whose name contains foo and whose license does not contain bar. Then new API would warrant a new function.

What's your motivation for creating a new function? How would you explain why search for a summary belongs to `search()` and not to `list()`.
